pretyping/reductionops.ml

```diff
@@ -1673,19 +1684,24 @@ let infer_cmp_universes _env pb s0 s1 univs =
 
 let infer_convert_instances ~flex u u' (univs,cstrs as cuniv) =
   if flex then
-    if UGraph.check_eq_instances univs u u' then cuniv
-    else raise NotConvertible
+    if UGraph.check_eq_instances univs u u' then Result.Ok cuniv
+    else Result.Error None
   else
     let qcstrs, cstrs' = UVars.enforce_eq_instances u u' Sorts.QUConstraints.empty in
-    let () = if not (Sorts.QConstraints.trivial qcstrs) then raise NotConvertible in
-    (univs, Constraints.union cstrs cstrs')
+    if Sorts.QConstraints.trivial qcstrs then
+      Result.Ok (univs, Constraints.union cstrs cstrs')
+    else
+      Result.Error None
 
 let infer_inductive_instances cv_pb variance u1 u2 (univs,csts) =
   let qcsts, csts' = get_cumulativity_constraints cv_pb variance u1 u2 in
-    let () = if not (Sorts.QConstraints.trivial qcsts) then raise NotConvertible in
-  (UGraph.merge_constraints csts' univs, Univ.Constraints.union csts csts')
+  if Sorts.QConstraints.trivial qcsts then
+    match UGraph.merge_constraints csts' univs with
+    | univs -> Result.Ok (univs, Univ.Constraints.union csts csts')
+    | exception (UGraph.UniverseInconsistency err) -> Result.Error (Some err)
+  else Result.Error None
 
-let inferred_universes : (UGraph.t * Univ.Constraints.t) universe_compare =
+let inferred_universes : (UGraph.t * Univ.Constraints.t, 'err) universe_compare =
```

Comment:
```suggestion
let inferred_universes =
```
we already have the type in the mli, it's not useful to duplicate it here
especially with a weak variable which misleads the reader into thinking it's polymorphic